### PR TITLE
Content/2516/release doc update

### PR DESCRIPTION
**Pull request by ddimatos**
##### SUMMARY
Documentation updates for the upcoming v1.2.0-beta.1 release.
Includes improved documentation
Includes all new module docs
Includes new playbook sample
Includes new JCL

##### ISSUE TYPE
- Docs Pull Request

### CODE REVIEW

**Comment by IBMAnsibleHelper** [comment]
Exception: ansible-test detected errors! 
 ERROR: Found 1 rstcheck issue(s) which need to be resolved:
ERROR: docs/source/modules/zos_mvs_raw.rst:49:0: Definition list ends without a blank line; unexpected unindent.
ERROR: The 1 sanity test(s) listed below (out of 41) failed. See error output above for details.

File: docs/source/playbooks.rst [hunk 1]
```diff
@@ -228,6 +228,28 @@ for Python and ``ZOAU`` have been added and used through the configuration.
...
+   PYZ: "/usr/lpp/rsusr/python36"
+   ZOAU: "/usr/lpp/IBM/zoautil"
+
+   environment_vars:
+     ZOAU_ROOT: "{{ ZOAU }}"
+     ZOAU_HOME: "{{ ZOAU }}"
+     PYTHONPATH: "{{ ZOAU }}/lib:{{ PYZ }}:/lib:/usr/lib"
+     _BPXK_AUTOCVT: "ON"
+     PATH: "{{ ZOAU }}/bin:/bin:/var/bin:{{ PYZ }}/bin"
+     LIBPATH: "{{ ZOAU }}/lib:{{ PYZ }}/lib:/lib:/usr/lib"
```
Comment (blakeinate) [inline docs/source/playbooks.rst:251]: Do we want to add `.` to the end here?

File: plugins/modules/zos_copy.py [hunk 2]
```diff
@@ -65,14 +65,14 @@
...
-      - The backup file name will be return on either success or failure of
+      - The backup file name will be returned on either success or failure of
         module execution such that data can be retrieved.
     type: bool
     default: false
     required: false
   backup_file:
     description:
-      - Specify the USS file name or data set name for the destination backup.
+      - Specify an unique USS file name or data set name for the destination backup.
```
Comment (blakeinate) [inline plugins/modules/zos_copy.py:75]: same comment as rst, not a big deal if we leave it

File: docs/source/modules/zos_copy.rst [hunk 3]
```diff
@@ -0,0 +1,625 @@
...
+  The backup file name will be returned on either success or failure of module execution such that data can be retrieved.
+
+
+  | **required**: False
+  | **type**: bool
+
+
+     
+backup_file
+  Specify an unique USS file name or data set name for the destination backup.
```
[the remark this answers is not in the document]
Comment (ddimatos) [reply docs/source/modules/zos_copy.rst:45]: It is not, i saw it too and figured i would pass on it but ill correct it now.

File: docs/source/playbooks.rst (hunk 1 above)
Comment (ddimatos) [reply docs/source/playbooks.rst:251]: Fixed, good catch, funny I caught this the other day for you :)

Comment (blakeinate) [reply docs/source/playbooks.rst:251]: I think they were different files, but your catch the other day led me to check this one too 😄

File: plugins/modules/zos_copy.py (hunk 2 above)
Comment (ddimatos) [reply plugins/modules/zos_copy.py:75]: Fixed.

File: docs/source/modules/zos_copy.rst [hunk 4]
```diff
@@ -0,0 +1,625 @@
...
+
+     
+mode
+  The permission of the destination file or directory.
+
+  If ``dest`` is USS, this will act as Unix file mode, otherwise ignored.
+
+  It should be kept in mind that modes are actually octal numbers. The user must either add a leading zero so that Ansible's YAML parser knows it is an octal number (like ``0644`` or ``01777``)or quote it (like ``'644'`` or ``'1777'``) so Ansible receives a string and can do its own conversion from string into number. Giving Ansible a number without following one of these rules will end up with a decimal number which will have unexpected results.
+
+  As of Ansible 1.8, the mode may be specified as a symbolic mode (for example, ``u+rwx`` or ``u=rw,g=r,o=r``).
```
[the remark this answers is not in the document]
Comment (asifmahmud) [reply docs/source/modules/zos_copy.rst:174]: @ddimatos @blakeinate I agree that specifying earlier Ansible version would be unnecessary. So, just having
```
The mode may also be specified as a symbolic mode (for example, ``u+rwx`` or ``u=rw,g=r,o=r``).
```
Would be enough

[the remark this answers is not in the document]
Comment (ddimatos) [reply docs/source/modules/zos_copy.rst]: Good catch @balkajbaf , worked it out with @asifmahmud and supplied an update.

File: docs/source/modules/zos_copy.rst [hunk 5]
```diff
@@ -0,0 +1,625 @@
...
+
+  As of Ansible 2.3, the mode may also be the special string ``preserve``.
```
[the remark this answers is not in the document]
Comment (ddimatos) [reply docs/source/modules/zos_copy.rst]: Good catch @balkajbaf , worked it out with @asifmahmud and supplied an update.

**Comment by IBMAnsibleHelper** [comment]
Exception: one or more steps failed during playbook execution, or host was unreachable! 
 
PLAY [all] *********************************************************************

TASK [Setting fact `data_set_name` for use by this sample] *********************
ok: [ec33025a.vmec.svl.ibm.com]

TASK [Fact `data_set_name` set with value] *************************************
ok: [ec33025a.vmec.svl.ibm.com] => {
    "msg": "OMVSADM.SOME.DATA.SET"
}

TASK [Detecting system name] ***************************************************
fatal: [ec33025a.vmec.svl.ibm.com]: UNREACHABLE! => {"changed": false, "msg": "Data could not be sent to remote host \"ec33025a.vmec.svl.ibm.com\". Make sure this host can be reached over ssh: Host key verification failed.\r\n", "unreachable": true}

PLAY RECAP *********************************************************************
ec33025a.vmec.svl.ibm.com  : ok=2    changed=0    unreachable=1    failed=0    skipped=0    rescued=0    ignored=0

**Pull request merged**